public/app/plugins/datasource/cloudwatch/utils/query/getStatsGroups.ts

```diff
@@ -0,0 +1,38 @@
+const antlr4 = require('antlr4');
+const ScrollQLLexer = require('./ScrollQLLexer').ScrollQLLexer;
+const ScrollQLParser = require('./ScrollQLParser').ScrollQLParser;
+const ScrollQLParserListener = require('./ScrollQLParserListener').ScrollQLParserListener;
+
+class GroupListener extends ScrollQLParserListener {
+  groupNames: string[] = [];
+
+  enterLogStats(ctx: any) {
+    this.groupNames = [];
+    if (ctx.groups && ctx.groups.length > 0) {
+      const groups = ctx.groups;
+
+      groups.forEach((group: any) => {
+        const proj = group.fieldSpec?.().proj;
+        if (proj) {
+          this.groupNames.push(proj.getText());
+        } else {
+          this.groupNames.push(group.getText());
+        }
+      });
+    }
+  }
+}
+
+export function getStatsGroups(text: string): string[] {
+  const queryPrefix = 'source test start=0 end=1|';
```

Comment:
What does this prefix do?

Reply:
Apparently internally all queries are prefixed with a `source` command, so a dummy prefix is prepended to queries before parsing so that the parser functions correctly.

Reply:
Interesting, probably just a comment about that here. Ideally if you can link to some code or docs where you found out about it.

Reply:
Had to ask jared at amazon about it, so I'm not sure if there are any docs anywhere, unfortunately.